### PR TITLE
Add yesqa pre-commit

### LINKED ISSUE #13812: Add pre-commit hook yesqa?
### Description

Flags unnecessary ``noqa``, so they aren't added to the codebase going forward.
@WilliamJamieson, you have more recent experience with pre-commit + flake8. Would this be useful?

### Additional context

https://github.com/asottile/yesqa

### PR DESCRIPTION
**Pull request by nstarman**
It checks for unnecessary noqa statements.

Fixes #13812 

Signed-off-by: Nathaniel Starkman (@nstarman) <nstarkman@protonmail.com>

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [x] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [x] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label. Codestyle issues can be fixed by the [bot](https://docs.astropy.org/en/latest/development/workflow/development_workflow.html#pre-commit).
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

**Comment by nstarman** [comment]
Maybe this should even be merged before #13781 because this PR provides a better record of which statements are intrinsically unnecessary vs being ignored by the config change.

**Review by WilliamJamieson — APPROVED** [review]
I like these changes. I also agree that this should be merged before #13781.

File: astropy/wcs/wcs.py [hunk 1]
```diff
@@ -3065,7 +3065,7 @@ def reorient_celestial_first(self):
         the spectral axis, followed by any others.
         Assumes at least celestial axes are present.
         """
-        return self.sub([WCSSUB_CELESTIAL, WCSSUB_SPECTRAL, WCSSUB_STOKES, WCSSUB_TIME])  # Defined by C-ext  # noqa: F821 E501
+        return self.sub([WCSSUB_CELESTIAL, WCSSUB_SPECTRAL, WCSSUB_STOKES, WCSSUB_TIME])  # Defined by C-ext  # noqa: E501, F821
```
Comment (pllim) [inline astropy/wcs/wcs.py:3068]: Wow, noqa code in alphabetical order. And I thought I was nitpicky.

**Review by pllim — APPROVED** [review]
I guess it doesn't hurt but I hope you won't try to introduce GitHub Copilot next. 😝 

Thanks!

**Comment by nstarman** [comment]
> I guess it doesn't hurt but I hope you won't try to introduce GitHub Copilot next. 😝

What's GH Copilot 👀  ?

**Comment by nstarman** [comment]
Ohhh... 😆.

**Pull request merged**